java/vector/src/main/java/org/apache/arrow/vector/complex/impl/UnionExtensionWriter.java

```diff
@@ -0,0 +1,38 @@
+package org.apache.arrow.vector.complex.impl;
+
+import org.apache.arrow.vector.ExtensionTypeVector;
+import org.apache.arrow.vector.types.pojo.Field;
+
+public class UnionExtensionWriter extends AbstractFieldWriter {
```

Comment:
This class is currently more skeleton like as the concrete details of the `ExtensionWriter` interface will need to be nailed down.